src/DevelopmentServer.js

```diff
@@ -200,7 +200,13 @@ export class DevelopmentServer {
           // so we append them as response headers instead
           res.append('set-cookie', cookie);
         });
-        res.send(isBase64Encoded ? Buffer.from(body, 'base64') : body);
+        // For HEAD requests, preserve the Content-Length header if set
+        // Express's send() will override it otherwise
+        if (method === 'HEAD' && headers['content-length']) {
+          res.end();
+        } else {
+          res.send(isBase64Encoded ? Buffer.from(body, 'base64') : body);
+        }
```

Comment:
I wonder if we shouldn't `end()` the response for every request after `send()`. something like:

```suggestion
        const buf = isBase64Encoded ? Buffer.from(body, 'base64') : body;
        if (body.length > 0) {
          res.send(body);
        }
        res.end();
```